diff --git a/platform/linuxbsd/detect.py b/platform/linuxbsd/detect.py
index 36e149f2b4..2c87bf09f5 100644
--- a/platform/linuxbsd/detect.py
+++ b/platform/linuxbsd/detect.py
@@ -310,6 +310,14 @@ def configure(env: "Environment"):
 
     if env["x11"]:
         env.Append(CPPDEFINES=["X11_ENABLED"])
+        env.ParseConfig("pkg-config x11 --cflags --libs")
+        env.ParseConfig("pkg-config xcursor --cflags --libs")
+        env.ParseConfig("pkg-config xinerama --cflags --libs")
+        env.ParseConfig("pkg-config xext --cflags --libs")
+        env.ParseConfig("pkg-config xrandr --cflags --libs")
+        env.ParseConfig("pkg-config xrender --cflags --libs")
+        env.ParseConfig("pkg-config xi --cflags --libs")
+        env.ParseConfig("pkg-config xfixes --cflags --libs")
 
     if env["vulkan"]:
         env.Append(CPPDEFINES=["VULKAN_ENABLED"])
