diff --git a/SConstruct b/SConstruct
index b86ccf6..7a9ff29 100644
--- a/SConstruct
+++ b/SConstruct
@@ -15,7 +15,7 @@ try:
 except:
     # Default tools with no platform defaults to gnu toolchain.
     # We apply platform specific toolchains via our custom tools.
-    env = Environment(tools=["default"], PLATFORM="")
+    env = Environment(tools=["default"], PLATFORM="", ENV={"PATH" : os.environ["PATH"]})
 
 env.PrependENVPath("PATH", os.getenv("PATH"))
 
